### PR TITLE
Refactoring labels and additional labels support

### PR DESCRIPTION
**Pull request by nlamirault**

### CODE REVIEW

File: charts/opentelemetry-collector/templates/clusterrole.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ metadata:
   name: {{ include "opentelemetry-collector.clusterRoleName" . }}
   labels:
     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+    {{- if or .Values.standaloneCollector.enabled (eq .Values.mode "deployment") -}}
+    app.kubernetes.io/component: standalone-collector
+    {{- else if or (and .Values.agentCollector.enabled (not (eq .Values.mode "deployment"))) (eq .Values.mode "daemonset") -}}
+    app.kubernetes.io/component: agent-collector
```
Comment (dmitryax) [inline charts/opentelemetry-collector/templates/clusterrole.yaml:11]: The terms `standalone-collector` and `agent-collector` are discouraged after we moved to the mode setting. I would say that `app.kubernetes.io/component: opentelemetry-collector` and mode potentiall can be reflected in another label if there is a real need.

**Review by dmitryax — COMMENTED** [review]
Please provide reasoning why new labels are needed in the PR description or create an issue

File: charts/opentelemetry-collector/templates/clusterrole.yaml (hunk 1 above)
Comment (TylerHelmuth) [reply charts/opentelemetry-collector/templates/clusterrole.yaml:11]: Yes, `app.kubernetes.io/component : opentelemetry-collector` would be preferred in all resources.  If the mode (currently `deployment` and `daemonset`) also needs reflected in a label that would be a different label.

**Comment by nlamirault** [comment]
@dmitryax i make some new commits

**Pull request closed without merging**